components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/ServerSessionManagementServiceImpl.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.internal.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.juli.logging.Log;
+import org.apache.juli.logging.LogFactory;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.identity.application.authentication.framework.ServerSessionManagementService;
+import org.wso2.carbon.identity.application.authentication.framework.cache.SessionContextCache;
+import org.wso2.carbon.identity.application.authentication.framework.context.SessionContext;
+import org.wso2.carbon.identity.application.authentication.framework.internal.FrameworkServiceDataHolder;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+
+public class ServerSessionManagementServiceImpl implements ServerSessionManagementService {
+
+    private static Log log = LogFactory.getLog(ServerSessionManagementServiceImpl.class);
+
+    @Override
+    public boolean removeSession(String sessionId) {
+
+        if (StringUtils.isBlank(sessionId)) {
+            return false;
+        }
+        // Retrieve session information from cache in order to publish event
+        SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(sessionId);
+        terminateSession(sessionContext, sessionId);
+        return true;
+    }
+
+    @Override
+    public boolean removeMySession(String username, String tenantDomain, String sessionId) {
```

Comment:
What if we remove this API from this service and move this API to UserSessionManagementService

API could be something like,

    public boolean removeMySession(User authenticatedUser, String sessionId) {

Reply:
The admin service can consume this new API

Reply:
We already have following APIs for terminating the session by `sessionId` in `UserSessionManagementService`. 

- [ `terminateSessionBySessionId(String userId, String sessionId)`](https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/UserSessionManagementService.java#L75)
- [`terminateSessionBySessionId(User user, int idpId, String sessionId)`](https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/UserSessionManagementService.java#L114)

Reply:
`removeMySession` API removed from `ServerSessionManagementService` in bfcebcb